lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs

```diff
@@ -30,6 +30,7 @@ data V4 = V4
     metaPath :: Path,
```

Comment:
Can we remove this?

Reply:
Not really. This one's still used for the actual http request.

Reply:
We could remove metaCanonicalPath instead since I only really added it for making testing easier. Tests would have to grab the canonical path from the canonical request after an unlines, which seemed a bit clunky.

Reply:
I feel a bit unsure about it: Is the request assembled in this function the correct type that the test harness could call `canonicalPath` directly?

Reply:
How does this look? https://github.com/mwu/amazonka/pull/1

Reply:
I think it looks good.